File: etrade-next/src/lib/orderStatus.ts
export const ORDER_STATUS = {
  PLACED: 0,
  PREPARING: 1,
  SHIPPED: 2,
  DELIVERED: 3,
  REJECTED: 4,
  COMPLETED: 5,
} as const;

export function orderStatusLabel(status: number) {
  if (status === ORDER_STATUS.PLACED) return "Order placed";
  if (status === ORDER_STATUS.PREPARING) return "Preparing";
  if (status === ORDER_STATUS.SHIPPED) return "Shipped";
  if (status === ORDER_STATUS.DELIVERED) return "Delivered";
  if (status === ORDER_STATUS.REJECTED) return "Rejected";
  if (status === ORDER_STATUS.COMPLETED) return "Completed";
  return `Status: ${status}`;
}

File: etrade-flutter/lib/core/order_status.dart
/// Canonical order workflow codes — must match `etrade-next/src/lib/orderStatus.ts`.
abstract final class OrderStatus {
  static const int placed = 0;
  static const int preparing = 1;
  static const int shipped = 2;
  static const int delivered = 3;
  static const int rejected = 4;
  static const int completed = 5;

  static String label(int status) {
    switch (status) {
      case placed:
        return 'Order placed';
      case preparing:
        return 'Preparing';
      case shipped:
        return 'Shipped';
      case delivered:
        return 'Delivered';
      case rejected:
        return 'Rejected';
      case completed:
        return 'Completed';
      default:
        return 'Processing';
    }
  }

  /// Prefer numeric [status] when present so labels never disagree with the server.
  static String displayLabel({required int? status, String? statusText}) {
    final s = status;
    if (s != null && s >= placed && s <= completed) {
      return label(s);
    }
    final t = statusText?.trim();
    if (t != null && t.isNotEmpty) return t;
    return 'Unknown';
  }
}
